examples/src/main/java/io/numaproj/numaflow/examples/sink/simple/SimpleSink.java

```diff
@@ -1,21 +1,30 @@
 package io.numaproj.numaflow.examples.sink.simple;
 
 import io.numaproj.numaflow.sink.Response;
+import io.numaproj.numaflow.sink.SinkDatumStream;
 import io.numaproj.numaflow.sink.SinkFunc;
 import io.numaproj.numaflow.sink.SinkServer;
 import io.numaproj.numaflow.sink.v1.Udsink;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.logging.Logger;
 
 public class SimpleSink {
     private static final Logger logger = Logger.getLogger(SimpleSink.class.getName());
 
-    private static Response[] process(Udsink.Datum[] datumList) {
-        Response[] responses = new Response[datumList.length];
-        for (int i = 0; i < datumList.length; i++) {
-            logger.info(datumList[i].getValue().toStringUtf8());
-            responses[i] = new Response(datumList[i].getId(), true, "");
+    private static List<Response> process(SinkDatumStream datumStream) {
+        ArrayList<Response> responses = new ArrayList<Response>();
+
+        while (true) {
+            Udsink.Datum datum = datumStream.ReadMessage();
+            // null indicates the end of the input
+            if (datum == null) {
```

Comment:
if we choose to pick a enum instead of null, we should change it here.

Reply:
it is a simple log sink